### PR TITLE
MANTA-4628 Rename extant buckets components such that the name reflects the function

### PR DESCRIPTION
**Pull request by kellymclaughlin**

### CODE REVIEW

File: lib/buckets/objects/metadata/update.js [hunk 1]
```diff
@@ -84,7 +84,8 @@ function updateObjectMetadata(req, res, next) {
 
         var metadataLocation = req.metadataPlacement.getObjectLocation(owner,
             bucket.id, bucketObject.name);
-        var client = req.metadataPlacement.getBorayClient(metadataLocation);
+            var client =
```
Comment (trentm) [inline lib/buckets/objects/metadata/update.js]: nit: indentation

File: lib/buckets/common.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ var format = util.format;
 var errors = require('../errors');
 
 var LIST_LIMIT = 1024;
-var BORAY_MULTIPLIER = 2;
+var BUCKETS_MDAPI_MULTIPLIER = 2;
 
 /*
  * A valid bucket name is composed of one or more "labels," separated by
```
Comment (trentm) [inline lib/buckets/common.js:25]: nit: there is a "boray" ref in a comment on line 70 here.

File: lib/buckets_mdapi_client.js [hunk 3]
```diff
@@ -41,38 +40,29 @@ function createClient(options, callback) {
...
-
-        if (clientArray.length === pnodes.length) {
-            // write ready cookie when clients have connected
-            log.info('all boray clients instantiated writing ready cookie');
-            try {
-                fs.writeFileSync('/var/tmp/electric-boray-ready', null);
-            } catch (e) {
-                throw new verror.VError(e, 'unable to write ready cookie');
-            }
-        }
```
Comment (trentm) [inline lib/buckets_mdapi_client.js:71]: This was cruft?

File: package.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
         "assert-plus": "^1.0.0",
         "backoff": "2.3.0",
         "bignum": "0.13.0",
-        "boray": "git+https://github.com/joyent/node-buckets-mdapi.git#v0.3.0",
+        "buckets-mdapi": "git+https://github.com/joyent/node-buckets-mdapi.git#MANTA-4628",
```
Comment (trentm) [inline package.json]: You will update this before merging?

File: lib/buckets_mdapi_client.js (hunk 3 above)
Comment (kellymclaughlin) [reply lib/buckets_mdapi_client.js:71]: Yeah, that was carryover from when this module lived in `electric-boray`/`buckets-mdplacement`.

File: package.json (hunk 4 above)
Comment (kellymclaughlin) [reply package.json]: Yep, for sure

File: lib/buckets/common.js (hunk 2 above)
Comment (kellymclaughlin) [reply lib/buckets/common.js:25]: Fixed

File: lib/buckets/objects/metadata/update.js (hunk 1 above)
Comment (kellymclaughlin) [reply lib/buckets/objects/metadata/update.js]: Fixed

**Review by trentm — DISMISSED** [review]
LGTM, modulo the  package.json dep update.

**Review by bowrocker — APPROVED** [review]
LGTM

**Review by trentm — APPROVED** [review]

**Pull request merged**